numba_extensions/numba_extensions/bf16/torch_interop.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import warnings
+import numpy as np
+from numba.np import numpy_support  # noqa: E402
+
+import numba.cuda.dispatcher
+from numba.cuda.dispatcher import _LaunchConfiguration
+
+from numba_extensions.bf16 import nv_bfloat16
+
+try:
+    import torch
+
+    _TORCH_IMPORTED = True
+except ImportError:
+    _TORCH_IMPORTED = False
+
+try:
+    from ml_dtypes import bfloat16  # noqa: F401 E402
+
+    _ML_DTYPES_IMPORTED = True
+except ImportError:
+    if _TORCH_IMPORTED:
+        warnings.warn(
+            "Pytorch installation detected. To use Numba-extensions' bfloat16"
+            "bindings, please also install ml_dtypes via `python -m pip install ml_dtypes`"
+        )
+
+    _ML_DTYPES_IMPORTED = False
+
+_WRAP_BF16_TENSOR = _TORCH_IMPORTED and _ML_DTYPES_IMPORTED
+
+
+def patch_numba():
```

Comment:
It would be good to log when patching Numba, like ptxcompiler and other Numba-patching libraries do, e.g. 

https://github.com/rapidsai/ptxcompiler/blob/ae91f8dad5e5ff7103128eaf1a77b923887334ec/ptxcompiler/patch.py#L261

Reply:
Added a logger to log.